internal/services/network/virtual_network_peering_data_source_test.go

```diff
@@ -0,0 +1,93 @@
+				check.That(data.ResourceName).Key("allow_forwarded_traffic").HasValue("true"),
+				check.That(data.ResourceName).Key("allow_gateway_transit").HasValue("false"),
+				check.That(data.ResourceName).Key("only_ipv6_peering_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("peer_complete_virtual_networks_enabled").HasValue("true"),
+				check.That(data.ResourceName).Key("use_remote_gateways").HasValue("false"),
+				check.That(secondResourceName).Key("allow_virtual_network_access").HasValue("true"),
+				check.That(secondResourceName).Key("allow_forwarded_traffic").HasValue("true"),
+				check.That(secondResourceName).Key("allow_gateway_transit").HasValue("false"),
+				check.That(secondResourceName).Key("only_ipv6_peering_enabled").HasValue("false"),
+				check.That(secondResourceName).Key("peer_complete_virtual_networks_enabled").HasValue("true"),
+				check.That(secondResourceName).Key("use_remote_gateways").HasValue("false"),
+			),
+		},
+	})
+}
+
+func (r VirtualNetworkPeeringDataSource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRg-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_virtual_network" "test-1" {
+  name                = "acctestvnet1-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  address_space       = ["10.0.1.0/24"]
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_virtual_network" "test-2" {
+  name                = "acctestvnet2-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  address_space       = ["10.0.2.0/24"]
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_virtual_network_peering" "test-1" {
+  name                         = "peer1to2"
+  resource_group_name          = azurerm_resource_group.test.name
+  virtual_network_name         = azurerm_virtual_network.test-1.name
+  remote_virtual_network_id    = azurerm_virtual_network.test-2.id
+  allow_virtual_network_access = true
+}
+
+resource "azurerm_virtual_network_peering" "test-2" {
+  name                         = "peer2to1"
+  resource_group_name          = azurerm_resource_group.test.name
+  virtual_network_name         = azurerm_virtual_network.test-2.name
+  remote_virtual_network_id    = azurerm_virtual_network.test-1.id
+  allow_virtual_network_access = true
+}
+
+data "azurerm_virtual_network_peering" "test-1" {
+  name                 = "peer1to2"
+  virtual_network_name = azurerm_virtual_network.test-1.name
+  resource_group_name  = azurerm_resource_group.test.name
+}
+
+data "azurerm_virtual_network_peering" "test-2" {
+  name                 = "peer2to1"
+  virtual_network_name = azurerm_virtual_network.test-2.name
+  resource_group_name  = azurerm_resource_group.test.name
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
```

Comment:
Can we prefix these with `acctest` so that it's clear what resources were created by our acceptance tests and so that our automation can clean these up in case of any failures
```suggestion
  name                         = "acctestpeer1to2"
  resource_group_name          = azurerm_resource_group.test.name
  virtual_network_name         = azurerm_virtual_network.test-1.name
  remote_virtual_network_id    = azurerm_virtual_network.test-2.id
  allow_virtual_network_access = true
}

resource "azurerm_virtual_network_peering" "test-2" {
  name                         = "acctestpeer2to1"
  resource_group_name          = azurerm_resource_group.test.name
  virtual_network_name         = azurerm_virtual_network.test-2.name
  remote_virtual_network_id    = azurerm_virtual_network.test-1.id
  allow_virtual_network_access = true
}

data "azurerm_virtual_network_peering" "test-1" {
  name                 = "acctestpeer1to2"
  virtual_network_name = azurerm_virtual_network.test-1.name
  resource_group_name  = azurerm_resource_group.test.name
}

data "azurerm_virtual_network_peering" "test-2" {
  name                 = "acctestpeer2to1"
  virtual_network_name = azurerm_virtual_network.test-2.name
  resource_group_name  = azurerm_resource_group.test.name
}
`, data.RandomInteger, data.Locations.Primary)
}
```